src/statecharts/service.cljc

```diff
@@ -10,31 +10,30 @@
   (add-listener [this id listener])
   (reload [this fsm]))
 
-(declare attach-fsm-scheduler)
-
 (defn wrap-listener [f]
   (fn [_ _ old new]
     (f old new)))
 
-(deftype Service [^:volatile-mutable fsm
-                  state
+(deftype Service [fsm
+                  store
                   ^:volatile-mutable running
                   clock
                   transition-opts]
   IService
   (start [this]
     (when-not running
       (set! running true)
-      (set! fsm (attach-fsm-scheduler this fsm))
-      (reset! state (impl/initialize fsm))
-      @state))
+      (store/initialize store fsm nil)))
   (send [_ event]
-    (reset! state (impl/transition fsm @state event transition-opts))
-    @state)
+    (let [old-state (store/get-state store nil)]
+      (store/transition store fsm old-state event transition-opts)))
   (add-listener [_ id listener]
-    (add-watch state id (wrap-listener listener)))
+    ;; Kind of gross to reach down into store's internals. Then again, the fact
+    ;; that the store a lone-store is an implementation detail known only to this
+    ;; namespace.
+    (add-watch (:state* store) id (wrap-listener listener)))
   (reload [this fsm_]
-    (set! fsm (attach-fsm-scheduler this fsm_))
+    ;; TODO: this is now a no-op. Remove from protocol?
```

Comment:
I might be missing something—I don't use services so I don't fully understand how they're integrated into an app—but I don't think that'll work. Since `fsm` is no longer a volatile you can't bash a new machine onto an existing service. You could make `reload` return a _new_ service, which the user would have to persist themselves, in a var or an atom. But if they're going to do that, they may as well reinitialize with `(service fsm)`.

I didn't notice this until now, but whatever we decide will have to be reflected in [the docs](https://lucywang000.github.io/clj-statecharts/docs/integration/re-frame/#integrate-with-the-service-api).

One thing this brings up, which the original code didn't handle either, is... does an old scheduler need to be shutdown, by unscheduling all of its handlers?

Reply:
I've thought about this a bit more... My design intuition says that you should avoid volatiles if you can. But, in this case, I'm not very invested. If the purpose of a service is to be a long-lived wrapper around state, but which can be updated with revised events, then I can see the argument for `fsm` being a volatile. I'll switch it back to keep `reload` working as it did.

My main goal was to break the circular dependency between schedulers and machines, which this PR still accomplishes. In the revised code, the `fsm` will be a volatile so that new events can be installed into the service, not to maintain a circular dependency between the fsm and its scheduler.

I'm not going to try to resolve whether `reload` should unschedule outdated events. That's a separate issue.

Reply:
> I'm not going to try to resolve whether reload should unschedule outdated events. That's a separate issue.

Thanks! Totally agree.